pex/bin/pex.py

```diff
@@ -503,11 +503,18 @@ def make_relative_to_root(path):
   return os.path.normpath(path.format(pex_root=ENV.PEX_ROOT))
 
 
-def main():
+def main(args=None, log_callback=None):
+  if log_callback is not None:
```

Comment:
looking through this, updating log in main() doesn't actually help since several other methods are using log as a global as well.  I'll update the code later with an alternative that allows you to update the logger for all methods.